src/System.Collections.Specialized/src/System/Collections/Specialized/NameObjectCollectionBase.cs

```diff
@@ -112,10 +112,10 @@ private void Reset(int capacity)
             _version++;
         }
 
-        private NameObjectEntry FindEntry(string key)
+        private NameObjectEntry? FindEntry(string? key)
         {
             if (key != null)
-                return (NameObjectEntry)_entriesTable[key];
+                return (NameObjectEntry)_entriesTable[key]!;
```

Comment:
maybe use `as` instead
```return _entriesTable[key] as NameObjectEntry;```

Reply:
I think that would be better, you would avoid the `!`.

Reply:
No, stick with the explicit cast.  It has different semantics.

Reply:
But you shouldn't need the `!`... just change the cast to be `(NameObjectEntry?)`

Reply:
Ah right. Good call.